examples/exex/minimal/src/main.rs

```diff
@@ -50,3 +51,45 @@ fn main() -> eyre::Result<()> {
         handle.wait_for_node_exit().await
     })
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{pin::pin, sync::Arc};
+
+    use reth::providers::{BundleStateWithReceipts, Chain};
+    use reth_exex::{ExExEvent, ExExNotification};
+    use reth_exex_test_utils::{test_exex_context, PollOnce, TestExExContext};
+
+    #[tokio::test]
+    async fn exex() -> eyre::Result<()> {
+        // Initialize a test Execution Extension context with all dependencies
+        let TestExExContext { ctx, genesis, provider_factory: _, mut events_rx, notifications_tx } =
+            test_exex_context().await?;
```

Comment:
I think we want a wrapper type for anything besides the `ctx` so that we add helper functions, like emit chain committed and assert functions for receiving etc

Reply:
yeah good idea